### DISCUSSION TITLE
Enable SIMD in query engine

### DISCUSSION THREAD

**Discussion by v0y4g3r** [GreptimeTeam/greptimedb#3652] [Ideas]
### What type of enhancement is this?

Performance

### What does the enhancement do?

Enabling SIMD feature in arrow/datafusion related crates to exploit vectorized execution.

- https://docs.rs/crate/arrow/43.0.0/features

### Implementation challenges

We need to do a thorough benchmark on execution with SIMD enabled/disabled to see how much we can gain from it.

**Comment by tisonkun** [comment]
Moved to Ideas until we have more concrete proposals.
